src/coreclr/System.Private.CoreLib/src/System/Threading/Monitor.CoreCLR.cs

```diff
@@ -68,8 +146,35 @@ private static void ThrowLockTakenException()
         **             SynchronizationLockException if the current thread does not
         **             own the lock.
         =========================================================================*/
-        [MethodImpl(MethodImplOptions.InternalCall)]
-        public static extern void Exit(object obj);
+        public static void Exit(object obj)
+        {
+            LeaveHelperAction exitBehavior = Exit_FastPath(obj);
+
+            if (exitBehavior == LeaveHelperAction.None)
+                return;
+
+            Exit_Slowpath(obj, exitBehavior);
```

Comment:
Can Exit_FastPath do anything when returning exitBehavior != LeaveHelperAction.None (note NOT EQUAL) that would break the subsequent fall-through to Exit_Slowpath (e.g. we did the fast-path exit, but it needs some other LeaveHelperAction... but we've done the Exit already)?

(same question on line 174)

Reply:
No.